### ISSUE TITLE
Option to support overwirte files when uploading files with the same name that already exist

### ISSUE THREAD

**Issue by jinxsong123** [altugbakan/rs-tftpd#6]
Now the rust-tftpd not support uploading files with the same name that already exist.
if uploading the file that already exist, the client will receive one error:  Error code 6: requested file already exists
Could you help to add one option to support this feature?

**Comment by stappersg** [comment]
On Sat, Oct 07, 2023 at 11:38:59PM -0700, jinxsong wrote:
> Now the rust-tftpd not support uploading files with the same name that
> already exist.

Something I call a design decision.
Correcting myself: Something I call a good design decision.


> if uploading the file that already exist, the client will receive one
> error:  Error code 6: requested file already exists

Nice.


> Could you help to add one option to support this feature?

Answering that question as

}  Could be considered to support overwriting existing files?
 
Yes, very shortly.  TFTP has no authentication, meaning every client
can destroy, also known as overwriting, existing files.

Current implemenation keeps files integer.



I see three possible ways out for the "problem":

* TFTP client picks another filename upon recieving
  the "Error code 6: requested file already exists"
* at TFTP server  move (or proces+delete) uploaded files
  Such proces watches a special "upload directory", so
  files outside that directory are not effected.
* The wanted feature as an option that is default disabled


I think that last option is what original poster is hoping for.
Time will tell how many true stakeholders show up.


Regards
Geert Stappers

P.S.
Free software is about libre software, not about gratuit.
-- 
Silence is hard to parse

**Comment by jinxsong123** [comment]
Thank you very much for your quickly reply
Very glad you could consider my suggestion, as it will give users more choices. Thanks again

Another small comment is about the 'tftpd --help' show below,  The description is not very accurate

![图片](https://github.com/altugbakan/rs-tftpd/assets/52711267/2bc4a856-22b9-40ba-95fc-e1ae5c735f66)

**Comment by altugbakan** [comment]
This is a good suggestion, thanks a lot! I will add a flag for supporting overwriting files, as even I use the same file for testing and debugging tftpd itself!
